procedures/crates.md

```diff
@@ -0,0 +1,146 @@
+in `rust-lang`) and that the experience is uniform across `rust-lang/rust` and these crates.
+
+### When should parts of the compiler be extracted into an out-of-tree crate?
+This is left to the discretion of compiler team members but should be discussed with the rest of
+the team, either through raising the question at the weekly triage meeting or asynchronously using
+an FCP in an issue. If the crate is a product of a working group, there should already be agreement
+within the working group that an out-of-tree crate is suitable.
+
+When considering creating an out-of-tree crate, it is worth balancing how general the crate should
+be with the increased maintenance burden that this may bring if widely used.
+
+### Where should compiler crates live?
+Out-of-tree compiler crates should be hosted in the `rust-lang` organization - this simplifies
+integration with external infrastructure tooling and will inherit existing team permissions on
+GitHub. It should be made clear in any documentation that the compiler team and any appropriate
+working groups are responsible for the crate. It is not recommended to start with a prototype in
+another organization or personal repository.
+
+### Can existing out-of-tree crates from personal accounts or other organizations be transferred?
+Yes, this is encouraged. In order to do this, discuss this with the team and familiarize yourself
+with [the GitHub documentation for repository transfers][repo_transfers] and then arrange to perform
+the transfer. Once a transfer is complete, a redirect will exist in the original account or
+organization and this will conflict with the names of any new forks of the transferred repository -
+an email to GitHub is required to resolve this.
+
+[repo_transfers]: https://help.github.com/en/articles/transferring-a-repository
+
+### Who owns these crates?
+It is desired that a compiler team member or working group has loose ownership over a crate so
+that there are clear owners who are responsible for making sure that new versions are published and
+that pull requests are reviewed.
+
+### What should these crates be named?
+Crate naming will be discussed when new out-of-tree crates are proposed to the compiler team.
+
+Crate naming will differ on a case-by-case basis. Crates that are inherently tied to the
+compiler would benefit from a name that is prefixed with `rustc_`. This is an indicator of how
+stable the crate may be to prospective users. Other crates, which are more general-purpose, will
+have names that are disentangled from the compiler.
+
+### Are there any limitations on the review policy for out-of-tree crates?
+Generally, the working groups and team members that are primarily free to maintain the crate using
+whatever practices are best suited to their group, however, there are some limitations so that there
+is some uniformity across the compiler and out-of-tree crates:
+
+- Every compiler team member must be able to review and approve PRs.
```

Comment:
Note that there are [a lot of other people](https://github.com/rust-lang/rust-central-station/blob/master/homu.toml.template) who have r+ rights than the compiler team on rust-lang/rust, for example, the entire language team has r+ in the repo. If we start moving to a out-of-tree crates model under this review policy then presumably a lot of the people in the RCS list would no longer have r+? Is this intentional and why?

Reply:
(I'm not the person who'd make the final call on this, these are just my thoughts)

I'm not sure how this is any different than other repositories that are out-of-tree, such as cargo, clippy or libc. Only the relevant teams/people have r+ on those repositories. I'm sure there'd be no issue with someone from another team being granted r+ in an out-of-tree crate should it make sense for that to happen. 

It is probably the case that the only reason there isn't a more fine-grained notion of who has r+ on what is because it's hard to do with everything in a single repository. I recognize that there are advantages to everyone with r+ on the repo being able to approve things (e.g. anyone can retry a PR if they notice it has spuriously failed) but I think that "only the team responsible for the crate" is a sane default.

Reply:
rust-lang/rust is a bit special in the sense that "the relevant teams" is much less clear cut because cargo, clippy, libc are all different programs or shippables while the out-of-tree crates I think we are discussing here combine to a single program "rustc". For example, the language team has the say over stabilization of many features and should be able to r+ those PRs.

More importantly, a language team member should be able to r- a PR or at least inform (with e.g. labeling) the compiler-team member who is reviewing a PR that "This PR should not be merged without T-Lang's approval".  This becomes more important when you can change semantics by simply bumping a crates version number. I think changes would be substantially harder to check in this model as compared to the relative ease I have today to skim almost all PRs for problematic things. These are my main concerns here which may perhaps be addressed in other ways...